client/src/components/UserRegisterForm.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { Button, Form, Grid, Header, Message } from "semantic-ui-react";
+
+export class UserRegisterForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: "",
+      name: "",
+      password: "",
+      confirmationPassword: "",
+      userSignUp: false
+    };
+  }
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  };
+  handleSubmit = e => {
+    e.preventDefault();
+    const { password, confirmationPassword } = this.state;
+    if (password !== confirmationPassword) {
+      alert("passwords dont match");
+    } else {
+      const userRequest = {
+        method: "POST",
+        body: JSON.stringify({
+          email: this.state.email,
+          name: this.state.name,
+          password: this.state.password,
+          confirmationPassword: this.state.confirmationPassword
+        }),
+        headers: {
+          "Content-Type": "application/json"
+        }
+      };
+      fetch("http://localhost:4000/auth/register", userRequest)
+        .then(res => res.json())
+        .then(this.resetForm);
+    }
+  };
+  resetForm = () => {
+    this.setState({
+      name: "",
+      email: "",
+      password: "",
+      confirmationPassword: "",
+      userSignUp: true
+    });
+  };
+  clickLogin = () => {
+    window.location.href = "/login";
+  };
+  render() {
+    let userSignUp;
+    if (this.state.userSignUp) {
+      userSignUp = "Your account has been successfully created.";
+    } else {
+      userSignUp = null;
+    }
```

Comment:
The variable `userSignUp` is not needed anymore I think as you use the Message component below. Then this can be removed.

Reply:
okay